components/push/src/subscriber.rs

```diff
@@ -101,19 +100,19 @@ impl PushManager {
     }
 
     // XXX: maybe -> Result<()> instead
-    pub fn unsubscribe(&self, channel_id: Option<&str>) -> Result<bool> {
+    pub fn unsubscribe(&mut self, channel_id: Option<&str>) -> Result<bool> {
         if self.conn.uaid.is_none() {
             return Err(ErrorKind::GeneralError("No subscriptions created yet.".into()).into());
         }
-        let uaid = self.conn.uaid.as_ref().unwrap();
+        let uaid = self.conn.uaid.as_ref().unwrap().to_owned();
```

Comment:
Out of curiosity, why is the `to_owned()` necessary here?

Reply:
Oh It's because I'm changing `ConnectHttp.unsubscribe` to take a mutable reference, so the borrow checker will complain because otherwise the `uaid` would be an immutable reference used in `delete_record`

In retrospect though, I can avoid the clone by moving this whole line into the `self.store.delete_record` so I'll do that 😄